src/vecs/collection.py

```diff
@@ -391,6 +391,48 @@ def _list_collections(cls, client: "Client") -> List["Collection"]:
                 xc.append(existing_collection)
         return xc
 
+    @classmethod
+    def _does_collection_exist(cls, client: "Client", name: str) -> bool:
+        """
+        PRIVATE
+
+        Checks if a collection with a given name exists within the database
+
+        Args:
+            client (Client): The database client.
+            name (str): The name of the collection
+
+        Returns:
+            Exists: Whether the collection exists or not
+        """
+
+        query = text(
+        f"""
+        select
+            relname as table_name,
+            atttypmod as embedding_dim
+        from
+            pg_class pc
+            join pg_attribute pa
+                on pc.oid = pa.attrelid
+        where
+            pc.relnamespace = 'vecs'::regnamespace
+            and pc.relkind = 'r'
+            and pa.attname = 'vec'
+            and not pc.relname ^@ '_'
+            and pc.relname = '{name}'
+        """
+        )
+
+        xc = []
+        with client.Session() as sess:
+            for name, dimension in sess.execute(query):
+                existing_collection = cls(name, dimension, client)
+                xc.append(existing_collection)
+        return len(xc) != 0
+
+
+
```

Comment:
rather than repeating this logic in a new method, can we use `get_collection` to check if it already exists?